### PR TITLE
add Smoke grenade and tear gas grenade

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

What should have been a long time ago, a smoke grenade for the syndicate and a tear gas grenade for sec.

**Media**

https://github.com/space-wizards/space-station-14/assets/60344369/3be6e814-3f40-45d9-9506-87cd64d0a6e1

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: VigersRay
- add: Added smoke and tear grenade.

### CODE REVIEW

File: Resources/Textures/Objects/Weapons/Grenades/smokegrenade.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Grenades/smokegrenade.rsi/meta.json:4]: Why is it empty?

File: Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json:4]: ditto

**Review by ficcialfaint — CHANGES_REQUESTED** [review]

File: Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+using Content.Shared.Chemistry.Components;
+using Robust.Shared.Audio;
+
+namespace Content.Server.Fluids.EntitySystems;
+
+[RegisterComponent]
+[Access(typeof(SmokeSystem))]
+public sealed class SmokeOnTriggerComponent : Component
```
Comment (0x6273) [inline Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs:8]: Why is this needed? Can't you use `ChemicalPayloadComponent` with smoke chems?

File: Resources/Textures/Objects/Weapons/Grenades/smokegrenade.rsi/meta.json (hunk 1 above)
Comment (VigersRay) [reply Resources/Textures/Objects/Weapons/Grenades/smokegrenade.rsi/meta.json:4]: What to write there?

Comment (ficcialfaint) [reply Resources/Textures/Objects/Weapons/Grenades/smokegrenade.rsi/meta.json:4]: > What to write there?

Idk, `Created by <your GH nick> (GitHub)`

File: Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs (hunk 3 above)
Comment (VigersRay) [reply Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs:8]: I'm not sure that it is possible to implement this through ChemicalPayload. In addition, the grenade should not be able to replace what is inside. If you want me to use ChemicalPayload, you must provide a couple of examples of entities in which a filled chemical charge is already installed, I did not find such in the game code.

File: Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by Cthulhuwu (Discord)",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json:4]: If you are using someone's Discord as a copyright, you should contain their discrim

File: Resources/Textures/Objects/Weapons/Grenades/smokegrenade.rsi/meta.json (hunk 1 above)
Comment (ficcialfaint) [reply Resources/Textures/Objects/Weapons/Grenades/smokegrenade.rsi/meta.json]: forgot to mention, that if it actually was made by you

File: Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs (hunk 3 above)
Comment (deltanedas) [reply Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs:8]: pretty sure you are able to:
1. use chemical payload
2. use solutioncontainermanager to initialize it with a smoke mix

since the crowbaring bit is on modular grenade's main component if you dont have that you wont have it do anything

File: Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json (hunk 4 above)
Comment (VigersRay) [reply Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json:4]: discri-what?

File: Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs (hunk 3 above)
Comment (VigersRay) [reply Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs:8]: I can not do it

And in PayloadCaseComponent there is no way to specify what will be inside.
image

File: Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json (hunk 4 above)
Comment (ficcialfaint) [reply Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json:4]: Four digits after #

Comment (VigersRay) [reply Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json:4]: > Four digits after #

After the discord update, the usernames are no longer numbered...

Comment (ficcialfaint) [reply Resources/Textures/Objects/Weapons/Grenades/tearsmokegrenade.rsi/meta.json:4]: 1. It's not out yet
2. Discord will still support the discrims for some time afaik

File: Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs (hunk 3 above)
Comment (0x6273) [reply Content.Server/Fluids/EntitySystems/SmokeOnTriggerComponent.cs:8]: `ChemicalPayloadComponent` uses item slots so you'd have probably to create prototypes for beakers filled with the chems. Or change `ChemicalPayloadComponent` to just use solution directly.

**Review by mirrorcult — CHANGES_REQUESTED** [review]
this should be using a chem payload like the rest of the grenades. its not really an issue if people can replace whats in it

**Comment by VigersRay** [comment]
> this should be using a chem payload like the rest of the grenades. its not really an issue if people can replace whats in it

And then what's the point of making a copy of the modular grenade but with a different name and sprite?

**Comment by mirrorcult** [comment]
because these would have prefilled contents?

**Comment by malchanceux** [comment]
> Wouldn't it be easier then to just put a regular modular grenade in the uplink and the sb belt that will be filled?
Wow free modular grenade in uplink real.

Jokes aside, you cant crack open a frag grenade for example like you would with a modular grenade from sci. Uplink sells **traitor** items for a reason. Another difference is that your smoke grenade doesnt beep unlike a modular grenade.

Just don't allow extracting the payload from the grenade and its okay to use ChemicalPayloadComponent.
> since the crowbaring bit is on modular grenade's main component if you dont have that you wont have it do anything

**Comment by VigersRay** [comment]
Okay then, there is a problem in that if I use a payload, it is difficult or impossible for me to regulate the life and smoke density.  How do I even get the right amount of elements to create a thick smoke screen that will last 30 seconds?  Here I am without a clue.  And why can FleshOnTrigger exist but SmokeOnTrigger can't?

**Comment by malchanceux** [comment]
Since taking the payload out of the smoke grenade would be impossible I believe you can go goofy with the mixture (or make an exclusive chemical that you can finetune), alternatively make a "smoke payload" similarly to the "explosive payload" we already have in-game

**Comment by VigersRay** [comment]
> this should be using a chem payload like the rest of the grenades. its not really an issue if people can replace whats in it

So, after a little thought, I realized that using a chemical payload is not an option since no chemical reaction in the game produces a thick enough smoke that can exist for more than 30 seconds. The only option would be to create something like FlashPayload but SmokePayload using SmokeOnTrigger

**Comment by metalgearsloth** [comment]
Re-open when you come back to it.

**Comment by VigersRay** [comment]
> Re-open when you come back to it.

the fact is that I still think that the SmokeOnTriggerComponent should exist, in particular it will be needed for blob spores, but I think I will open a new pr where there will only be changes in the smoke code with support for colored smoke, and then I will open a separate pr for the smoke grenade. What about the tear grenade... Since you cut out the ability for smoke to transmit chemicals, it can no longer be implemented (((

**Comment by deltanedas** [comment]
:trollface:
